src/rgw/rgw_lc.cc

```diff
@@ -368,13 +343,14 @@ class LCObjsLister {
   string prefix;
   vector<rgw_bucket_dir_entry>::iterator obj_iter;
   rgw_bucket_dir_entry pre_obj;
+	uint64_t num_noncurrent{0};
   int64_t delay_ms;
 
 public:
   LCObjsLister(rgw::sal::Driver* _driver, rgw::sal::Bucket* _bucket) :
       driver(_driver), bucket(_bucket) {
     list_params.list_versions = bucket->versioned();
-    list_params.allow_unordered = true;
+    list_params.allow_unordered = bucket->versioned() ? false : true; // need ordering for NewerNoncurrentVersions
```

Comment:
are you sure we can't keep the unordered listing? it should preserve the ordering within a given index shard, which should be all we care about for ordering of versions

Reply:
no, you're presumably correct